mono_aot_register_module conflicting definitions

I work with a quite complex build system calling mkbundle with `--keep-temp`, manually patching the resulting C files (e.g. writing our own `main`) and then compiling the result with GCC 7.4.0. 

This whole process aims to allow us to link precisely and circumvent several problems (possible bugs) we encountered in mkbundle.

It's quite complicated to create a minimal setup reproducing this issue, the original script failing on mono 6.4 is here: https://github.com/renode/renode/blob/master/tools/packaging/make_linux_portable.sh

The problem I encounter when finally calling gcc:

```
/home/.../renode/tools/packaging/renode_1.8.0_portable-workdir/bundler.c:381:6: error: conflicting types for 'mono_aot_register_module'
 void mono_aot_register_module (void *aot_info);
      ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/.../renode/tools/packaging/renode_1.8.0_portable-workdir/bundler.c:5:0:
/usr/include/mono-2.0/mono/jit/jit.h:107:1: note: previous declaration of 'mono_aot_register_module' was here
 mono_aot_register_module (void **aot_info);
```

The `bundler.c` file is `temp.c` generated by mkbundle with our additions.

I do believe this was introduced with https://github.com/mono/mono/pull/15132 - before that we had to manually add a declaration for `mono_aot_register_module`.

I seems to me that a fix should be applied to `mcs/tools/mkbundle/template_common.inc`. There is this fragment:

```
#ifdef USE_DEFAULT_MONO_API_STRUCT
// We don't export in jit.h
// So declare here, and get it from mono
void mono_aot_register_module (void *aot_info);
#endif // USE_DEFAULT_MONO_API_STRUCT
```

The comment is clearly not relevant anymore and this code is pointed by the GCC error: bundler.c:381:6.

I will create a PR fixing this, but I am not 100% sure how to test it robustly, so please feel free to drop it if needed.

I believe the PR will just drop the quoted piece of code.

--- a/mcs/tools/mkbundle/bundle-mono-api.inc
+++ b/mcs/tools/mkbundle/bundle-mono-api.inc
@@ -19,7 +19,7 @@ typedef struct BundleMonoAPI
 	void (*mono_register_bundled_assemblies) (const MonoBundledAssembly **assemblies);
 	void (*mono_register_config_for_assembly) (const char* assembly_name, const char* config_xml);
 	void (*mono_jit_set_aot_mode) (MonoAotMode mode);
-	void (*mono_aot_register_module) (void* aot_info);
+	void (*mono_aot_register_module) (void** aot_info);
 	void (*mono_config_parse_memory) (const char *buffer);
 	void (*mono_register_machine_config) (const char *config_xml);
 } BundleMonoAPI;

--- a/mcs/tools/mkbundle/template_common.inc
+++ b/mcs/tools/mkbundle/template_common.inc
@@ -49,12 +49,6 @@ validate_api_struct ()
 	exit (1);
 }
 
-#ifdef USE_DEFAULT_MONO_API_STRUCT
-// We don't export in jit.h
-// So declare here, and get it from mono
-void mono_aot_register_module (void *aot_info);
-#endif // USE_DEFAULT_MONO_API_STRUCT
-
 static void
 init_default_mono_api_struct ()
 {

